### PR TITLE
Add note about lazy-loaded images with width or height 0 not loading.

### PR DESCRIPTION
**Pull request by Psychpsyo**
### Description

I've added a note to the documentation for  element's `loading` property that points out that images with a width or height of 0 are never loaded if it's set to `lazy`.

### Motivation

I ran into that issue and had to get this info from some random comment on Stack Overflow when really, it should've been here.

### Additional details

It seems like [the standard](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-loading) does not mention this but it is the behavior in at least Firefox and Edge, though I wasn't able to find bug reports for either of them. (if this even is a bug)
The behavior can easily be verified by saving this code snippet into an html file and opening it in a browser.
```html
<img src="https://resources.whatwg.org/logo.svg" loading="lazy" style="height: 0;">
```

### CODE REVIEW

**Comment by eeeps** [comment]
I think, per spec, they are supposed to load.

Reasoning:

1. The lazy loading spec is built on IntersectionObserver, and [initializes an observer](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#start-intersection-observing-a-lazy-loading-element) with [the default threshold of zero](https://w3c.github.io/IntersectionObserver/#dom-intersectionobserverinit-threshold).
2. [Zero-area IntersectionObserver targets are supposed to queue IntersectionObserverEntries](https://wpt.fyi/results/intersection-observer/same-document-zero-size-target.html)

Accordingly, I see [your test img](https://cdpn.io/pen/debug/bGQKvYq) loading in the network tabs of Firefox, Chrome, and Safari (but not painting, because its height is zero).

**Comment by Psychpsyo** [comment]
> Accordingly, I see [your test img](https://cdpn.io/pen/debug/bGQKvYq) loading in the network tabs of Firefox, Chrome, and Safari (but not painting, because its height is zero). [Screenshot](https://o.img.rodeo/q_auto/smdqvoqw30gilogpknrx.png).

You are right, I missed that. I've updated the PR and made a test case that more accurately shows the problem I was encountering:
https://codepen.io/Psychpsyo/pen/zYMJGRK
The image here is not getting loaded since its size (before loading) is 0 x 0, making it not intersect at all with the ``overflow: hidden`` div it's contained in. When removing the ``loading`` attribute, the image gets requested, stops being 0 x 0 and shows up on the page.

**Comment by estelle** [comment]
Pinging @eeeps

**Comment by eeeps** [comment]
@estelle @Psychpsyo sorry for the delay!

Ah, now I see the point. Your example shows it clearly. However I worry that the language is confusing and that this is an uncommon problem (although its helpful to explain the underlying mechanisms).

> intersect a visible part of an element at all

The meaning of intersect here is weird, and confusing, for all of the reasons we were confused earlier in the thread (how can something that's 0x0 intersect with anything?)

> (Since unloaded images have a width and height of 0.)

I was going to say that this should be specific and say `clientWidth` and `clientHeight`, but that's not true... for instance ``s that have `width` and `height` attributes often *do* have an initial non-zero clientHeight/width.

Putting `width` and `height` on lazyloaded images [fixes this issue](https://codepen.io/eeeps/pen/mdvxoMd) and is a best practice, [recommended by the spec](https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element:~:text=Developers%20are%20encouraged%20to%20specify%20a%20preferred%20aspect%20ratio) because it helps with other things, notably layout shifts. Perhaps this note should simply recommend those attributes, and your explanatory example could be incorporated into the [dedicated lazyloading article,](https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading#images_and_iframes) in order to explain the deeper concepts of threshholds and initial layout sizes of unloaded images?

**Comment by eeeps** [comment]
Also — hot off the presses — the spec and Chrome are both changing in a way that would load the image in the example, I think https://groups.google.com/a/chromium.org/g/blink-dev/c/cn9_443jGgI/m/Db5wnIhxAAAJ

File: files/en-us/web/html/element/img/index.md [hunk 1]
```diff
@@ -144,6 +144,7 @@ This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attrib
       - : Defers loading the image until it reaches a calculated distance from the viewport, as defined by the browser. The intent is to avoid the network and storage bandwidth needed to handle the image until it's reasonably certain that it will be needed. This generally improves the performance of the content in most typical use cases.
 
     > **Note:** Loading is only deferred when JavaScript is enabled. This is an anti-tracking measure, because if a user agent supported lazy loading when scripting is disabled, it would still be possible for a site to track a user's approximate scroll position throughout a session, by strategically placing images in a page's markup such that a server can track how many images are requested and when.
+    > **Note:** Images that have `loading` set to `lazy` will never be loaded if they do not intersect a visible part of an element at all, even if loading them would change that. (Since unloaded images have a width and height of 0.)
```
Comment (estelle) [inline files/en-us/web/html/element/img/index.md:147]:
```suggestion
    > **Note:** Images with `loading` set to `lazy` will never be loaded if they do not intersect a visible part of an element, even if loading them would change that as unloaded images have a `width` and `height` of `0`. Putting `width` and `height` on lazyloaded images fixes this issue and is a best practice, [recommended by the specification](https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element:~:text=Developers%20are%20encouraged%20to%20specify%20a%20preferred%20aspect%20ratio). Doing so also helps prevent layout shifts.
```

**Review by estelle — COMMENTED** [review]
I made a suggestion per all the comments in the thread.

**Review by estelle — APPROVED** [review]
Yay! Thanks! 🎉

**Pull request merged**